### ISSUE TITLE
Error when an HGNC ID lookup fails

### ISSUE THREAD

**Issue by johnbachman** [sorgerlab/indra#41]
In processing the results from the query generated by the following script:

```
from indra.biopax import biopax_api as ba
bp = ba.process_pc_neighborhood(['TP63'])
bp.get_phosphorylation()
```

The following error occurs:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Sending Pathway Commons query...
Pathway Commons query returned model...
indra/biopax/processor.py:251: UserWarning: Cannot handle complex enzymes.
  warnings.warn('Cannot handle complex enzymes.')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/p63_test.py in <module>()
      2 
      3 bp = ba.process_pc_neighborhood(['TP63'])
----> 4 bp.get_phosphorylation()

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/biopax/processor.pyc in get_phosphorylation(self, force_contains)
     53     def get_phosphorylation(self, force_contains=None):
     54         stmts = self._get_generic_modification('phospho', 
---> 55                                                force_contains=force_contains)
     56         for s in stmts:
     57             self.statements.append(Phosphorylation(*s))

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/biopax/processor.pyc in _get_generic_modification(self, mod_filter, mod_gain, force_contains)
    274             enzs = BiopaxProcessor._get_agents_from_entity(controller_pe)
    275             subs = BiopaxProcessor._get_agents_from_entity(input_spe,
--> 276                                                            expand_pe=False)
    277             for enz, sub in itertools.product(listify(enzs), listify(subs)):
    278                 # If neither the required enzyme nor the substrate is

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/biopax/processor.pyc in _get_agents_from_entity(bpe, expand_pe, expand_er)
    413         # If it is a single entity, we get its name and database
    414         # references
--> 415         name = BiopaxProcessor._get_element_name(bpe)
    416         db_refs = BiopaxProcessor._get_db_refs(bpe)
    417         agent = Agent(name, db_refs=db_refs, mods=mods, mod_sites=mod_sites)

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/biopax/processor.pyc in _get_element_name(bpe)
    503 
    504         # Canonicalize name
--> 505         name = re.sub(r'[^\w]', '_', name)
    506         if re.match('[0-9]', name) is not None:
    507             name = 'p' + name

/Users/johnbachman/.virtualenvs/p27env/lib/python2.7/re.pyc in sub(pattern, repl, string, count, flags)
    153     a callable, it's passed the match object and must return
    154     a replacement string to be used."""
--> 155     return _compile(pattern, flags).sub(repl, string, count)
    156 
    157 def subn(pattern, repl, string, count=0, flags=0):

TypeError: expected string or buffer
```

The immediate source of the error is that the string in the call to sub is None; however, the reason for this is that in the _get_element_name static method of the Biopax processor, the HGNC lookup fails. The HGNC ID reported by the call to _get_hgnc_id is 13473, but the subsequent call to _get_hgnc_name(hgnc_id) using that ID returns None, which is assigned to name, and is subsequently than passed to the regex.

The display name for the element causing the problem is "p63". A lookup at genenames.org indicates that the actual HGNC ID for TP63 is 15979.

Perhaps the right thing to do here is to simply check if the HGNC ID lookup fails, and if so, use the getDisplayName string instead?

We might want to keep a record of cases where the HGNC IDs have turned out to be erroneous so we can fix them.

**Comment by bgyori** [comment]
I already dealt with this problem for UniProt - tracking down deprecated entries. This is the same problem here, and entry being replaced by another one. The entry here:
http://www.genenames.org/cgi-bin/gene_symbol_report?hgnc_id=HGNC:13473
actually refers to TP63 (and I'm sure I can get some info on this reference from the API). 

I implemented reverting to the display name here: 19da21e25210c2ecc46c8ce0283da2cd5a4bc9b2 first, just in case.

**Comment by bgyori** [comment]
This is essentially fixed in 8db27ac8cac0d91b3ee80d291d256cd5ec02e881, one caveat is that when the offline hgnc_entries.txt file is not available in the indra/data folder, INDRA will use the HGNC web service, from which I haven't figured out how to get deprecated entries by ID.
